### DISCUSSION TITLE
TreeView item change background on click

### DISCUSSION THREAD

**Discussion by vadimbarda** [bdlukaa/fluent_ui#806] [Q&A] [answered]
Hi, community. Please help me to implement functionality to change node background on click,
I use onItemInvoked callback. Then in this callback I remove this node and add new one with changed background:
```
var siblings = node.parent?.children ?? rootItems;
if (siblings.contains(node)) {
  siblings[siblings.indexOf(node)] = newNode;
}
```
But I don't see changes in UI. But I see changes after I collapse and expand parent. Where I am wrong?
Thank you in advance

**Comment by bdlukaa** [comment]
Have you tried to `setState`?

**Reply by vadimbarda** [reply]
Yes, I did. There is tree like File Explorer with lazy loading. Each node is directory in file system. I implemented dynamic changing lead icon depends on collapse/expanded state. But background color changing on click doesn't work for me. It is not help for me to wrap node changing with setState like:
```
var siblings = node.parent?.children ?? rootItems;
   if (siblings.contains(node)) {
      setState(() {  
         siblings[siblings.indexOf(node)] = newNode;
      });
}
```
Maybe you have some link to example I can review? I investigated examples in this repository, but there is no something I need. Thank you

**Reply by vadimbarda** [reply]
Hi @bdlukaa Any suggestions?

**Reply by bdlukaa** [reply, marked as answer]
As a workaround, you can use a key `GlobalKey<TreeView>` and build the items programatically. I filed #810 to track this issue

**Reply by vadimbarda** [reply]
thank you
